File: tkpdcore/src/main/java/com/tokopedia/core/referral/presenter/IReferralPresenter.java
package com.tokopedia.core.referral.presenter;

import android.support.v4.app.FragmentManager;

/**
 * Created by ashwanityagi on 18/09/17.
 */

public interface IReferralPresenter {

    void initialize();

    void checkLoginAndFetchReferralCode();

    void shareApp(FragmentManager fragmentManager);

    void getReferralVoucherCode();

    void copyVoucherCode(String code);

    String getReferralSubHeader();

    String getHowItWorks();

    String getVoucherCodeFromCache();

    Boolean isAppShowReferralButtonActivated();

    String getReferralTitleDesc();
}

File: features/im/referral/src/main/java/com/tokopedia/referral/view/presenter/IReferralPresenter.kt
package com.tokopedia.referral.view.presenter

import android.support.v4.app.FragmentManager

/**
 * Created by ashwanityagi on 18/09/17.
 */

interface IReferralPresenter {

    val referralSubHeader: String

    val howItWorks: String

    val voucherCodeFromCache: String

    val isAppShowReferralButtonActivated: Boolean?

    val referralTitleDesc: String

    fun initialize()

    fun checkLoginAndFetchReferralCode()

    fun shareApp(fragmentManager: FragmentManager)

    fun getReferralVoucherCode()

    fun copyVoucherCode(code: String)
}

